mesonbuild/interpreter.py

```diff
@@ -2182,6 +2182,35 @@ def parse_default_options(self, default_options):
                 newoptions = [defopt] + self.environment.cmd_line_options.projectoptions
                 self.environment.cmd_line_options.projectoptions = newoptions
 
+    def set_backend(self):
+        if self.backend is not None:
+            return
```

Comment:
Why is this needed? A comment might be in order.

Reply:
When parsing subproject the backend is already set.

Reply:
This commit fix 3 things, maybe commit message wasn't clear:

1) `project(default_options : 'backend=msvc')` silently has no effect because the backend was already decided at that point. I agree it's corner case but I find it weird that it's the only option not behaving like all others. At the very least we should warn. But since I made the change already, better keep it IMO.
2) `project(default_options : 'backend_max_links=42')` silently has no effect, no backend-specific option gets parsed there.
3) `meson -Dbackend_max_links=42` silently has no effect. Actually the only way to set that option is with `meson configure`.

I think consistency is really important, for ALL options we must respect:
cmdline > subproject() default > project() default > meson_options.txt/builtin default